### PR TITLE
doc: Fix case in titles

### PR DESCRIPTION
**Pull request by FrancescoSer**
Unified case in titles.
Fixed small grammar issues.

Signed-off-by: Francesco Servidio <francesco.servidio@nordicsemi.no>

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -490,7 +490,7 @@ of 3 is explained below.
     };
 ```
 
-## [IMAGE TRAILERS](#image-trailers)
+## [Image Trailers](#image-trailers)
```
Comment (utzig) [inline docs/design.md:493]: Why "Trailers" capitalized here?

File: ext/fiat/README.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ does manage to build that file, but the work on that branch was never finished
 (the correctness proofs of implementation templates still apply, but the
 now abandoned prototype specialization facilities there are unverified).
 
-## Working With Fiat Crypto Field Arithmetic
+## Working with fiat-crypto field arithmetic
```
Comment (utzig) [inline ext/fiat/README.md:40]: This file comes from an external project and should not be changed, please revert.

**Review by utzig — CHANGES_REQUESTED** [review]

File: ext/fiat/README.md (hunk 2 above)
Comment (FrancescoSer) [reply ext/fiat/README.md]: fixed

File: docs/design.md (hunk 1 above)
Comment (FrancescoSer) [reply docs/design.md]: fixed

**Review by utzig — APPROVED** [review]
LGTM

**Review by nvlsianpu — APPROVED** [review]

**Review by gustavonihei — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Pull request merged**